From 19f7f8542891b7138523659e8662e74404c8bab5 Mon Sep 17 00:00:00 2001
From: Cyrill Gorcunov <gorcunov@openvz.org>
Date: Tue, 21 Apr 2009 15:58:39 +0400
Subject: [PATCH 84/86] net: bridge - process skbs has been already substituted due to via_phys_dev

When via_phys_dev is enabled we substitute skb->dev with
master_dev and pass it back to bridge code. Instead of
dropping such skb we should pass it up to network stack
to process.

Signed-off-by: Cyrill Gorcunov <gorcunov@openvz.org>
Signed-off-by: Pavel Emelyanov <xemul@openvz.org>
---
 net/bridge/br_input.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/net/bridge/br_input.c b/net/bridge/br_input.c
index 8809156..77fc3b0 100644
--- a/net/bridge/br_input.c
+++ b/net/bridge/br_input.c
@@ -166,7 +166,7 @@ struct sk_buff *br_handle_frame(struct net_bridge_port *p, struct sk_buff *skb)
 		/* fall through */
 	case BR_STATE_LEARNING:
 		if (skb->brmark == BR_ALREADY_SEEN)
-			return 0;
+			return skb;
 
 		out = p->br->via_phys_dev ? p->br->master_dev : p->br->dev;
 
-- 
1.6.3.1

